documentation/internal/Element-Type-Specs.md

```diff
@@ -457,6 +458,56 @@ _Allows user to build a list of items, such as an **Ingredients List**_
 
 ---
 
+<a name="search"/>
+
+### Search (Lookup)
+
+- **type/code**: `search`
+- **category**: `Question`
+
+_Live "search-as-you-type" lookups for user to find and select items_
+
+#### Input parameters
+
+- **label**: `string` -- as above
+- **description**: `string` -- as above [Optional]
+- **placeholder**: `string` -- text to show in search field before user input (default: "Search...")
+- **source**: `EvaluatorExpression` -- query to retrieve results for user selection. In order to make this dynamic in response to user input, there should be at least one input node that uses an `objectProperties` operator to fetch user input. In this case, user input is contained within a special object named `search`, and can be acquired with the following query node:
+
+  ```
+  {
+    operator: "objectProperties",
+    children: [ "search.text" ]
+  }
+  ```
+
+- **icon**: `string` -- the name of the icon shown in the search box, from Semantic-UI icons. (default: "search" : 🔍 )
+- **multiSelect**: `boolean` -- whether or not the user can select multiple items for their response (default: `false`)
+- **minCharacters**: `number` -- the minimum number of characters the user must type before the search query executes (default: 1). This is useful in situations where need the user to look up a specific item without being able to freely browse through the entire results list. For example, to look up organisation in our system using "registration" code, we set `minCharacters = 6`, so the user will need to know an exact code rather than being able to try characters one at a time.
+- **displayFormat**: `object` -- defines how to display the search results and the user's selection cards. See `displayFormat` for the [List Builder](#list-builder) (above) for detailed explanation. In this case, however, instead of a `code` substitution, the display string should contain property names from the result object. For example:
+  ```
+  displayFormat:  {
+                    title: "\${firstName} \${lastName}"
+                    description: "\${username}"
+                  }
+  ```
+  If not specified, a generic "default" display will be shown, using the first 1-2 properties on the result object.
+- **resultFormat**: `object` -- same as `displayFormat`, but used when specifying a format for the "result" display that is different to the selection card display. If not specified, `resultFormat` will just be the same as `displayFormat`.  
```

Comment:
Maybe needs to say "where" does this `resultFormat` will be displayed. That is in the result list after the user adds a valid input to search, right?

Reply:
Okay, I added a couple of screenshots to clarify: d643f1e2